Components/Metrics/PointToSurfaceDistance/elxPointToSurfaceDistanceMetric.h

```diff
@@ -0,0 +1,172 @@
+  using TransformType = typename Superclass1::TransformType;
+  using TransformPointer = typename Superclass1::TransformPointer;
+  using InputPointType = typename Superclass1::InputPointType;
+  using OutputPointType = typename Superclass1::OutputPointType;
+  using TransformParametersType = typename Superclass1::TransformParametersType;
+  using TransformJacobianType = typename Superclass1::TransformJacobianType;
+  using FixedImageMaskType = typename Superclass1::FixedImageMaskType;
+  using FixedImageMaskPointer = typename Superclass1::FixedImageMaskPointer;
+  using MovingImageMaskType = typename Superclass1::MovingImageMaskType;
+  using MovingImageMaskPointer = typename Superclass1::MovingImageMaskPointer;
+  using MeasureType = typename Superclass1::MeasureType;
+  using DerivativeType = typename Superclass1::DerivativeType;
+  using ParametersType = typename Superclass1::ParametersType;
+
+   /** Typedefs inherited from elastix. */
+  using ElastixType = typename Superclass2::ElastixType;
+  using ElastixPointer = typename Superclass2::ElastixPointer;
+  using ConfigurationType = typename Superclass2::ConfigurationType;
+  using ConfigurationPointer = typename Superclass2::ConfigurationPointer;
+  using RegistrationType = typename Superclass2::RegistrationType;
+  using RegistrationPointer = typename Superclass2::RegistrationPointer;
+  using ITKBaseType = typename Superclass2::ITKBaseType;
+  using FixedImageType = typename Superclass2::FixedImageType;
+  using MovingImageType = typename Superclass2::MovingImageType;
+
+  /** The fixed image dimension. */
+  itkStaticConstMacro( FixedImageDimension, unsigned int,  FixedImageType::ImageDimension );
+
+  /** The moving image dimension. */
+  itkStaticConstMacro( MovingImageDimension, unsigned int, MovingImageType::ImageDimension );
+
+  using PointSetType = FixedPointSetType;
+  using ImageType = typename Superclass1::ImageType;
+
+
+  /** Sets up a timer to measure the initialisation time and
+   * calls the Superclass' implementation.
+   */
+  virtual void Initialize();
+
+  /**
+   * Do some things before all:
+   * \li Check and print the command line arguments fp and mp.
+   *   This should be done in BeforeAllBase and not BeforeAll.
+   */
+  virtual int BeforeAllBase();
+
+  /**
+   * Do some things before registration:
+   * \li Load and set the pointsets.
+   */
+  virtual void BeforeRegistration();
+
+/** Function to read the corresponding points. */
+/** Function to read the corresponding points. */
+  using ImageConstPointer = typename Superclass1::ImageType::ConstPointer;
+
+  unsigned int ReadLandmarks( const std::string & landmarkFileName, typename PointSetType::Pointer & pointSet, typename FixedImageType::ConstPointer image );
+
+  /** Overwrite to silence warning. */
+  virtual void SelectNewSamples( void ){}
+
+protected:
+
+  /** The constructor. */
+  PointToSurfaceDistanceMetric(){}
+  /** The destructor. */
+  virtual ~PointToSurfaceDistanceMetric() {}
```

Comment:
You may consider C++11 `= default` instead of an empty function body `{}`, for both the default-constructor and the destructor.

Reply:
I missed this